app/_hub/kong-inc/openid-connect/index.md

```diff
@@ -124,3145 +129,3071 @@ kong_version_compatibility:
-the plugin tries to find value from:
+1. we want to only use the authorization code flow and the session authentication.
+2. we want to set the response mode to `form_post` so that authorization codes won't get logged to the access logs.
+3. we want to preserve the original request query arguments over the authorization code flow redirections.
+3. we want to redirect the client to original request url after the authorization code flow so that
+   the `POST` request (because of `form_post`) is turned to the `GET` request, and the browser address bar is updated
+   with the original request query arguments.
+4. we don't want to include any tokens in the browser address bar.
 
-* header `X-Client-Config` or `Client-Config`
-* query argument `client_config`
-* post argument `client_config`
+```bash
+http -f patch :8001/plugins/5f35b796-ced6-4c00-9b2a-90eef745f4f9 \
+  config.auth_methods=authorization_code                         \
+  config.auth_methods=session                                    \
+  config.response_mode=form_post                                 \
+  config.preserve_query_args=true                                \
+  config.login_action=redirect                                   \
+  config.login_tokens=
+```
+```http
+HTTP/1.1 200 OK
+```
+```json
+{
+    "id": "5f35b796-ced6-4c00-9b2a-90eef745f4f9",
+    "name": "openid-connect",
+    "service": {
+        "id": "5fa9e468-0007-4d7e-9aeb-49ca9edd6ccd"
+    },
+    "config": {
+        "auth_methods": [
+            "authorization_code",
+            "session"
+        ],
+        "login_action": "redirect",
+        "preserve_query_args": true,
+        "login_tokens": null
+    }
+}
+```
 
-The value stored in the parameter can be `integer` or actual
-`client_id` found in `config.client_id`. In case it is an integer,
-that is used to pick up correct parameters by index from the following:
-
-* `config.client_id`
-* `config.client_secret`
-* `config.redirect_uri`
-* `config.login_redirect_uri`
-* `config.logout_redirect_uri`
-
-Indexing starts from `1`. If no such index is found, then the index of `1`
-is used. You can also use the actual `client_id` and then we use the index
-of the `config.client_id` for the rest of the parameters (and fallback to `1` if
-not found in specified index).
-
-
-#### config.redirect_uri
+#### Test the Authorization Code Flow
 
-Specify where you want the browser to be redirected (from IdP) in
-case of a successful login. The ID Token will be appended
-as an URL fragment to the redirect URL. If this parameter
-is not specified, the ID Token will be returned instead
-of redirect. This as `client_id` and `client_secret` can
-also be an array if different clients use different
-redirect uris.
+1. Open Service Page with some query arguments:
+   ```bash
+   open http://service.test:8000/?hello=world 
+   ```
+   <img src="/assets/images/docs/openid-connect/authorization-code-flow-1.png">
+   
+2. See that the browser is redirected to Keycloak login page:
```

Comment:
```suggestion
2. See that the browser is redirected to Keycloak login page. Enter the username (john) & password(doe) for the user we created previously.
```

Reply:
In my case Keycloack immediately told me to update my password after logging in, which I did.

Reply:
I see now that Keycloak has a setting called "temporary" which is what triggers the password reset. It should be indicated on the instructions/screenshots for creating the user that this setting should be set to false.